### DISCUSSION TITLE
Difference between CloseToTargetObjective and TargetObjective

### DISCUSSION THREAD

**Discussion by setuhn** [experimental-design/bofire#388] [Q&A] [answered]
Hello,
I have discovered BoFire through a presentation given by Prof. Ruth Misener and have been exploring this tool since then. I originally employed Ax for simple BO tasks but I would like to transition to BoFire for its ease of use and syntax clarity. However, I am not an expert in BO or BoTorche and I fail to understand the difference between CloseToTargetObjective and TargetObjective. Especially since CloseToTargetObjective seems to count as an objective and can be used in MOBO while TargetObjective counts as a constraint if I understand correctly and therefore doesn't count as an actual objective? Moreover, they both seem to define their output differently with a simple abs function for CloseToTargetObjective  and a sigmoid function for TargetObjective. Maybe it is my lack of knowledge of BO or BoTorche that hinders my understanding here. I would be very grateful if someone could clarify the subject for me. What are the differences and which should be used in what circumstances?

**Comment by R-M-Lee** [comment, marked as answer]
Hi there, sorry no-one noticed your question before.

Both TargetObjective and CloseToTargetObjective are valid obejctives that can be used in either multi- or single-objective problems. The difference is that CloseToTarget rewards getting closer to the target no matter how far away you are. TargetObjective is the product of two sigmoidal functions (one of which is 'flipped horizontally') so has a flat region whose size is governed by the tolerance parameter (and steepness to some extent). 

So if your goal is to really get as close as possible to some target number, then use CloseToTarget. But if you are actually fine with getting to within +/- some tolerance, then use Target.

Hopefully this helps.

**Comment by setuhn** [comment]
Hi there,

Thank you very much for your detailed answer! This is exactly what I wanted
to know.

Cheers!

Le mer. 21 août 2024 à 12:32, Robert Lee ***@***.***> a
écrit :
